code/modules/mob/living/damage_procs.dm

```diff
@@ -57,7 +58,10 @@
 		if(AGONY)
 			halloss += effect // Useful for objects that cause "subdual" damage. PAIN!
 		if(IRRADIATE)
-			radiation += max(effect * ((100-run_armor_check(null, "rad", "Your clothes feel warm.", "Your clothes feel warm."))/100),0)//Rads auto check armor
+			var/rad_damage = effect
+			if(blocked != -1) // Setting blocked to -1 overrides radiation armor checks, which are automatic otherwise

```

Comment:
Blocked cannot possibly equal negative 1. The above return statement, `if(!effect || (blocked <= 0)) return 0` prevents that.